flink-table/flink-table-planner/src/main/java/org/apache/calcite/avatica/util/DateTimeUtils.java

```diff
@@ -1039,6 +1141,46 @@ public String getFraction() {
 			return fraction;
 		}
 	}
+
+	/** Deals with values of {@code java.time.OffsetDateTime} without introducing
+	 * a compile-time dependency (because {@code OffsetDateTime} is only JDK 8 and
+	 * higher). */
+	private interface OffsetDateTimeHandler {
+		boolean isOffsetDateTime(Object o);
+		String stringValue(Object o);
```

Comment:
Do we really need this interface `stringValue` ? The two implementations: one is static and one throws an exception, i didn't see why we should keep this.

Reply:
this is copied from avactica 1.14.0 release